### ISSUE TITLE
early stopping

### ISSUE THREAD

**Issue by jdbravo** [ankandrew/fast-plate-ocr#32]
I’m trying to train a model for Colombian license plates, but I keep encountering this error:
Epoch 51: early stopping

Additionally, in every epoch, I see the message:
“val_plate_acc did not improve from 0.00000”

I have the dataset and all required files in this repository: https://gitlab.com/jdbravo/plates-ocr-train .

What am I doing wrong?

Thank you!

**Comment by ankandrew** [comment]
Hi! Try the following:

1. Checkout the following branch from this repo: `MobileViTV2-backbone`
2. Download the following weights file from this repo [global_mobile_vit_v2_ocr.keras](https://github.com/ankandrew/fast-plate-ocr/releases/download/arg-plates/global_mobile_vit_v2_ocr.keras).
3. To your `train.sh` add `--weights-path` param and point to that Keras file you downloaded.

This was you are fine-tuning on the model that has seen lots of plates from different countries, because if training from scratch as you are, the model might need more data to learn effectively.

Also, I suggest to monitor training with tensorboard option `--tensorboard` and play around with the learning rate, since you are fine-tuning.

**Comment by ankandrew** [comment]
@jdbravo Were you able to train? Let me know how it went.

**Comment by jdbravo** [comment]
@ankandrew Sorry for the delay! Yes, license plates in Colombia have 6 slots. I think I can’t change the number of slots in the config file from 7 to 6 if I’m using the global_mobile_vit_v2_ocr.keras weights file. That was causing me some early stops problems.

How many images would be a good starting point if I want to train a model from scratch without using the base model global_mobile_vit_v2?

Thank you!

**Comment by ankandrew** [comment]
The `max_plate_slots` for the `global_mobile_vit_v2` is set to 9 and not 7. Even if that is set to 9 during training it has seen many plates with len of 4, 5, 6, ... 9 the `global_mobile_vit_v2`. EarlyStopping would happen when model is not learning, I suspect something weird happened during training with your script. See below colab, with your dataset, on the validation set I fine-tuned `global_mobile_vit_v2` model and got plate accuracy of 0.923, https://colab.research.google.com/drive/1DzDUORZeqJqJzQvmMy1DeMIQ51Rd5KoJ.

More data would be more helpful, but you can just export the model to `.onnx` and use it with `fast-plate-ocr` with existing codebase. I attach the model in case you want to use that. Feel free to explore and with more data the fine-tune process.

[global_mobile_vit_v2_colombia.keras.zip](https://github.com/user-attachments/files/18818544/global_mobile_vit_v2_colombia.keras.zip)

**Comment by jdbravo** [comment]
@ankandrew I just updated the repo with more photos. Feel free to use them if you want to add the Colombian plate set to your global_mobile_vit_v2 model.

In Colab, I noticed that you set the number of epochs to 30. What method do you use to determine this value?

The images I’m using are cropped inside the plate, showing only the numbers without the plate frame. Is this okay, or could it cause any issues?

**Comment by ankandrew** [comment]
> I just updated the repo with more photos. Feel free to use them if you want to add the Colombian plate set to your global_mobile_vit_v2 model.

You gave me an idea to start a community section of models, and could init with your contribution that section 👍

> In Colab, I noticed that you set the number of epochs to 30. What method do you use to determine this value?

You can try fine-tuning for more epochs, the number `30` I used is just an arbitrary number. EarlyStopping would stop if val plate accuracy doesn't improve for some time, so you can just put a bigger number as number of epochs if you want.

> The images I’m using are cropped inside the plate, showing only the numbers without the plate frame. Is this okay, or could it cause any issues?

No problem with that, as long as the plate doesn't get very outside boundaries it won't cause any issue. I think you did the same, but I didn't add translation augmentation due to the nature of your dataset.
